src/codegen/ruby/java_types.rb

```diff
@@ -26,12 +26,14 @@
   'array of snowflakes' => 'ImmutableList<Snowflake>',
   'array of strings' => 'ImmutableList<String>',
   'array of two integers (current_size, max_size)' => 'ImmutableList<Long>',
+  'array of two integers (shard_id, num_shards)' => 'ImmutableList<Long>',
```

Comment:
Is an ImmutableList a good idea here? Or should we instead maybe move the Shard class to Data.

Reply:
ImmutableList is consistent with the other array of two integers there. We can still use the Shard class and convert to ImmutableList<Integer> before it's sent.

Reply:
Where in the discord docs does it use that terminology. I find it very dangerous considering Lists aren't meant to just contain 2 values or really any set number. They're more defined as a list of things when here that really isnt the case, it's a pair of items.

Reply:
Everywhere we have arrays, we use an ImmutableList. Raw arrays don't work as well with modern java (e.g., with generics and streams).

It's true that we could use a Pair or some other type, but it's going to be appear in JSON as any other array does.

Reply:
We shouldn't fudge it, we can make a custom deserializer for the Shard and have the Shard class move over to Data. It really isnt an array either it's just displayed that way.

Reply:
Added classes to handle these